### PR TITLE
Adds connectivity check on categories

### PR DESCRIPTION
**Pull request by samagra14**
## Description

This PR makes the CategoryActivity check for Internet connection and then only load categories. Once the offline caching is complete we may add an optional feature of loading the cached categories.

## Related issues and discussion
##993
 
 ## Screen-shots, if any
28584474_1611570842264226_647152436_o

 
 ## Checklist
 
Make sure you've done all the following (You can delete the checklist before submitting)
 
 - [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.
 - [x] Code is well documented
 - [ ] Include unit tests for new functionality
 - [x] All user-visible strings are made translatable
 - [x] Code passes Travis builds in your branch
 - [x] If you have multiple commits please combine them into one commit by squashing them.
 - [x] Read and understood the contribution guidelines .

### CODE REVIEW

**Comment by samagra14** [comment]
@teolemon @Karljoones Please review this PR. I have resolved merge conflicts and added connectivity check in the Categories activity.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java [hunk 1]
```diff
@@ -16,10 +18,28 @@ public static Intent getIntent(Context context) {
...
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_category);
+        checkNetworkAndShowScreen();
+    }
+
+    private void checkNetworkAndShowScreen(){
+        ConnectivityManager cm =
+                (ConnectivityManager)this.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
```
Comment (Karljoones) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java]: For code maintainability, can you use the Utils.isNetworkConnected? Just to cut down on rewriting of code :)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java [hunk 2]
```diff
@@ -16,10 +18,28 @@ public static Intent getIntent(Context context) {
...
+        boolean isConnected = activeNetwork != null &&
```
Comment (sachincool) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java]: There is a  Helper Method in Utils.utils   named as `isNetworkConnected(Context context)`   You could use that and reduce this further.

**Comment by Karljoones** [comment]
Can you provide us with a screenshot of the layouts that you changed?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java (hunk 1 above)
Comment (sachincool) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java]: Lol.. Did we both just wrote the same thing at same time ?

**Comment by samagra14** [comment]
@Karljoones I have made the required changes.

**Comment by sachincool** [comment]
I think this PR also Solves https://github.com/openfoodfacts/openfoodfacts-androidapp/issues/692

**Comment by sachincool** [comment]
Wow, I didn't think it through Nice @Karljoones.

**Comment by sachincool** [comment]
Wow, I didn't think it through Nice @Karljoones.

**Comment by samagra14** [comment]
@Karljoones @teolemon I have made all the required changes as suggested by you.

**Comment by sachincool** [comment]
Just add the check if the list is empty or not  and then make the no internet dialog appear respectively.  Great Finding @ManavPoddar  you can work on it.

**Comment by samagra14** [comment]
@sachincool @ManavPoddar @Karljoones 
This has to be changed regarding memory issues as discussed in #1039. Loading an entire list and saving it is a really bad idea if the list can be fetched from the server. I was hoping to change this sort of implementation in #1039.

**Comment by sachincool** [comment]
@samagra14 No, that is why recycler view was introduced.  Recycle the views.  the activity should just be recycled. we could store the list in onSavedInstance bundle, that way it will restore it in screen rotation aswell as coming back from mainactivity. fetching the list multiple times from 1 device could be bandwitdh  taking for both server and device. What's your thoughts? @Karljoones @ManavPoddar 
what i want to do is just simply tell android to not kill that specific activity till the app closes.

**Pull request merged**